client/post-editor/media-modal/image-editor/image-editor-canvas.jsx

```diff
@@ -127,6 +134,41 @@ const MediaModalImageEditorCanvas = React.createClass( {
 		MediaUtils.canvasToBlob( newCanvas, callback, this.props.mimeType, 1 );
 	},
 
+	/**
+	 * Compare initial `transform` properties with the given `transform` object
+	 * gotten from `props` component object and computes if the image has been transformed.
+	 *
+	 * @return {Boolean} `true` ist the image has changed
+	 */
+	hasTransformed() {
+		return JSON.stringify( defaults.transform ) !== JSON.stringify( this.props.transform );
+	},
+
+	/**
+	 * Compare the initial `crop` object with the given `crop` object
+	 * gotten from `props` component object.
+	 * Since that the window is defined through of events of the user interface
+	 * and sometimes gets tricky get back to the initial value
+	 * we add a threshold to detect when user wanna undo its changes.
+	 *
+	 * @return {Boolean} `true` is the image has been cut
+	 */
+	hasCropped() {
```

Comment:
Actually we can just read `hasChanges`. This should make things a lot simpler
https://github.com/Automattic/wp-calypso/blob/master/client/state/ui/editor/image-editor/reducer.js#L47